alchemist-engine/src/test/kotlin/it/unibo/alchemist/core/test/TestSimulationControl.kt

```diff
@@ -51,7 +53,7 @@ class TestSimulationControl : FreeSpec(
                     time shouldBe firstJump
                     goToTime(nextJump)
                     play()
-                    workerPool.submit { run() }
+                    delay(awaitForNextJump) // best solution so far for now, you cannot be sure that the simulation is played again
```

Comment:
@DanySK I know that this is really fragile -- I can try with something more sophisticated, but at least for the test should be ok, what do you think?

Reply:
Merging but taking your proposal for a fresh API as a promise :smile: